app/main.py

```diff
@@ -32,13 +41,13 @@ def main() -> None:
     ]
 
     # run the programs
-    service.run_program(wake_up_program)
-    service.run_program(sleep_program)
+    await service.run_program(wake_up_program)
+    await service.run_program(sleep_program)
```

Comment:
you cannot run all programs in parallel. What is play song will be run earlier than speaker is switched on?